lib/ecto/adapters/myxql/connection.ex

```diff
@@ -43,20 +43,28 @@ if Code.ensure_loaded?(MyXQL) do
     end
 
     @impl true
-    def to_constraints(%MyXQL.Error{mysql: %{name: :ER_DUP_ENTRY}, message: message}) do
+    def to_constraints(error, opts \\ [])
+    def to_constraints(%MyXQL.Error{mysql: %{name: :ER_DUP_ENTRY}, message: message}, opts) do
       case :binary.split(message, " for key ") do
-        [_, quoted] -> [unique: strip_quotes(quoted)]
+        [_, quoted] ->
```

Comment:
I believe we can make this simpler: (untested)

```elixir
[_, quoted] -> [unique: normalize_index_name(quoted, opts[:source])]
```

and then:

```elixir
defp normalize_index_name(quoted, source) do
  name = strip_quotes(quoted)

  if source do
    String.trim_leading(name, "#{source}.")
  else
    name
  end
end
```

what do you think?

Reply:
If one table name is enough i would take this. Your suggested function name is much better i will change the function name.

Reply:
As it works with the first table only. I have changed the implementation to your suggested normalize_index_name function.